pkg/rpc/server/server.go

```diff
@@ -1688,12 +1806,23 @@ func (s *Server) getFakeNextBlock() (*block.Block, error) {
 // witness invocation script in case of `verification` trigger (it pushes `verify`
 // arguments on stack before verification). In case of contract verification
 // contractScriptHash should be specified.
-func (s *Server) runScriptInVM(t trigger.Type, script []byte, contractScriptHash util.Uint160, tx *transaction.Transaction, verbose bool) (*result.Invoke, *response.Error) {
-	b, err := s.getFakeNextBlock()
-	if err != nil {
-		return nil, response.NewInternalServerError("can't create fake block", err)
+func (s *Server) runScriptInVM(t trigger.Type, script []byte, contractScriptHash util.Uint160, tx *transaction.Transaction, b *block.Block, verbose bool) (*result.Invoke, *response.Error) {
```

Comment:
We can't keep the signature the same, because we need to distinguish between historic and simple calls. Currently it is done by checking that the passed block is nil.